File: src_tc/2dmap/cross.c



   /* calculate a X b return new vector in c */




void cross(a,b,c)

double a[],
       b[],
       c[];

   {
  
    c[1] = a[2] * b[3] - a[3] * b[2]; 
    c[2] = a[3] * b[1] - a[1] * b[3];
    c[3] = a[1] * b[2] - a[2] * b[1];
      
    return;
   }

File: unsupported/asf_ops_R2.1.2/ssp/pp_correlator/cross.f
c SccsId = @(#)cross.f	2.41 3/24/98


	subroutine cross(x1,y1,z1,x2,y2,z2,u1,u2,u3)

        implicit none

        character*128 SccsId_cross
        data SccsId_cross
     +  /'@(#)PPcross.f:2.41'/


	real*8 u1,z2,u2,r,u3,y1,x1,z1,y2,x2

	u1=y1*z2-z1*y2
	u2=z1*x2-x1*z2
	u3=x1*y2-y1*x2
	r=(u1**2+u2**2+u3**2)**.5
	u1=u1/r
	u2=u2/r
	u3=u3/r
	return
	end
